### PR TITLE
Support packet filtering for VPCS

### PR DESCRIPTION
**Pull request by noplay**
Ref https://github.com/GNS3/gns3-gui/issues/765

We can add support other emulators after merging this branch. The code can handle case where the node type doesn't support filters.

### CODE REVIEW

File: gns3server/compute/base_node.py [hunk 1]
```diff
@@ -586,6 +586,30 @@ def _add_ubridge_udp_connection(self, bridge_name, source_nio, destination_nio):
...
+    def _ubridge_apply_filters(self, bridge_name, filters):
+        """
+        Apply filter like rate limiting
+
+        :param bridge_name: bridge name in uBridge
+        :param filters: Array of filter dictionnary
+        """
+        yield from self._ubridge_send('bridge reset_packet_filters ' + bridge_name)
+        i = 0
+        for (type, values) in filters.items():
```
Comment (grossmj) [inline gns3server/compute/base_node.py]: Careful, type is a Python keyword.

File: gns3server/compute/vpcs/vpcs_vm.py [hunk 2]
```diff
@@ -388,6 +392,16 @@ def port_add_nio_binding(self, port_number, nio):
...
 
     @asyncio.coroutine
+    def port_update_nio_binding(self, port_number, nio):
+        if not self._ethernet_adapter.port_exists(port_number):
+            raise VPCSError("Port {port_number} doesn't exist in adapter {adapter}".format(adapter=self._ethernet_adapter,
+                                                                                           port_number=port_number))
+        if self.ubridge:
+            yield from self._update_ubridge_udp_connection("VPCS-{}".format(self._id), self._local_udp_tunnel[1], nio)
+        elif self.is_running():
+            raise VPCSError("Sorry, adding a link to a started VPCS instance is not supported without using uBridge.")
```
Comment (grossmj) [inline gns3server/compute/vpcs/vpcs_vm.py]: Should say 'updating a link'.

File: gns3server/controller/link.py [hunk 3]
```diff
@@ -26,6 +26,68 @@
...
+                "minimum": 0,
+                "maximum": 32767,
+                "unit": "ms"
+            }
+        ]
+    },
+    {
+        "type": "corrupt",
+        "name": "Corrupt",
+        "description": "The percentage represents the chance for a packet to be corrupt",
```
Comment (grossmj) [inline gns3server/controller/link.py]: to be corrupted.

**Comment by noplay** [comment]
Feedback apply

**Pull request merged**